File: python/hashing/LC_3487_MaximumUniqueSubarraySumAfterDeletion.py
# https://leetcode.com/problems/maximum-unique-subarray-sum-after-deletion
from typing import List


class Solution:
    def maxSum(self, nums: List[int]) -> int:
        unique = set()
        max_sum = 0
        max_ele = float("-inf")
        for num in nums:
            if num >= 0 and num not in unique:
                max_sum += num
                unique.add(num)
            elif num < 0 and not unique:
                max_ele = max(num, max_ele)

        return max_sum if unique else max_ele

File: java/src/hashing/LC_3487_MaximumUniqueSubarraySumAfterDeletion.java
package hashing;

import java.util.HashSet;

public class LC_3487_MaximumUniqueSubarraySumAfterDeletion {

    public int maxSum(int[] nums) {
        HashSet<Integer> unique = new HashSet<>();
        int maxSum = 0, maxEle = Integer.MIN_VALUE;
        for (int num : nums) {
            if (num >= 0 && !unique.contains(num)) {
                unique.add(num);
                maxSum += num;
            } else if (num < 0 && unique.isEmpty()) {
                maxEle = Math.max(maxEle, num);
            }
        }
        return unique.isEmpty() ? maxEle : maxSum;
    }
}
